src/Razor/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -330,11 +330,36 @@ public static CodeWriter WriteAutoPropertyDeclaration(this CodeWriter writer, IL
             writer.Write(" ");
             writer.Write(propertyName);
             writer.Write(" { get; set; }");
+            writer.WritePropertyValue(typeName, defaultValue);
             writer.WriteLine();
 
             return writer;
         }
 
+        private static void WritePropertyValue(this CodeWriter writer, string typeName, string propertyValue)
+        {
+            if (propertyValue != null)
+            {
+                writer.Write(" = ");
+                switch (typeName)
+                {
+                    case "System.String":
```

Comment:
Do we do this in other places here (switch based on the type name). It feels a bit iffy

Reply:
There are places where we evaluate typeNames (for bools and strings). There are codepaths like `ToStringLiteral` and `ToBooleanLiteral` that conditional write out certain things depending on the type. However, this is the only place where we do a switch like this.